### PR TITLE
Update baby-llama.cpp

### PR DESCRIPTION
**Pull request by 0xspringtime**
Seems to be an error in the implementation of the operator!= function. It attempts to compare the this pointer (a llama_hparams_lora object) with the other pointer (a llama_hparams object) using memcmp. This can lead to incorrect results because the sizes of the objects being compared (sizeof(llama_hparams) and sizeof(llama_hparams_lora)) are different, should now be able to compare two llama_hparams_lora objects for inequality.

### CODE REVIEW

**Review by ggerganov — APPROVED** [review]
Thanks for spotting this - will wait for @xaedes to approve before merging

**Comment by 0xspringtime** [comment]
Got it, glad I could help in even a minor way.

**Review by xaedes — APPROVED** [review]

**Comment by 0xspringtime** [comment]
@ggerganov could we get a merge now?

**Pull request merged**
